### ISSUE TITLE
key import after key export panics on delegation keys

### ISSUE THREAD

**Issue by riyazdf** [docker/notary#1066]
After running `notary key export` on a delegation key, the delegation key gets PEM headers for `gun`, `path`, and `role`, example here:

```
gun:
path: tuf_keys/<KEY_ID>
role: user
```

`notary key import` will notice there's a path, but try to parse out a [gun from the path](https://github.com/docker/notary/blob/master/utils/keys.go#L171) and panic:

```
panic: runtime error: slice bounds out of range
goroutine 1 [running]:
panic(0x920d40, 0xc420012090)
/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/docker/notary/utils.ImportKeys(0xe04e60, 0xc420020398, 0xc42027ee50, 0x1, 0x1, 0x7ffc9e3c7f1f, 0x4, 0x0, 0x0, 0xc420252e80, ...)
/go/src/github.com/docker/notary/utils/keys.go:114 +0x1378
main.(*keyCommander).importKeys(0xc420256a00, 0xc4202aa200, 0xc4201d4be0, 0x1, 0x5, 0x0, 0x0)
/go/src/github.com/docker/notary/cmd/notary/keys.go:425 +0x235
main.(*keyCommander).(main.importKeys)-fm(0xc4202aa200, 0xc4201d4be0, 0x1, 0x5, 0x0, 0x0)
/go/src/github.com/docker/notary/cmd/notary/keys.go:102 +0x52
github.com/docker/notary/vendor/github.com/spf13/cobra.(*Command).execute(0xc4202aa200, 0xc4201d4af0, 0x5, 0x5, 0xc4202aa200, 0xc4201d4af0)
/go/src/github.com/docker/notary/vendor/github.com/spf13/cobra/command.go:561 +0x20c
github.com/docker/notary/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4200b7200, 0x0, 0x0, 0x0)
/go/src/github.com/docker/notary/vendor/github.com/spf13/cobra/command.go:651 +0x367
github.com/docker/notary/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4200b7200, 0xc4200b7200, 0xffffffffffffffff)
/go/src/github.com/docker/notary/vendor/github.com/spf13/cobra/command.go:610 +0x2b
main.main()
/go/src/github.com/docker/notary/cmd/notary/main.go:200 +0x82
```

cc @fermayo who found this issue

**Comment by endophage** [comment]
@riyazdf this is fixed by #1067 right?

**Comment by riyazdf** [comment]
yes - it didn't close automatically because it was merged to a branch instead of master

**Comment by riyazdf** [comment]
Fix released in `v0.4.3` and patch is carried to master
